packages/matter.js/src/codec/BtpCodec.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ByteArray, Endian } from "../util/ByteArray.js";
+import { DataReader } from "../util/DataReader.js";
+import { DataWriter } from "../util/DataWriter.js";
+
+export interface BtpHandshakeRequest {
+    version: ByteArray,
+    attMtu: number,
+    clientWindowSize: number,
+}
+
+export interface Header {
+    handshakeBit: number,
+    managementBit: number,
+    ackMsgBit: number,
+    endSegmentBit: number,
+    beginSegmentBit: number
+}
+
+export interface BtpHandshakeResponse {
+    version: ByteArray,
+    attMtu: number,
+    windowSize: number
+}
+
+export interface BtpPacketPayload {
+    ackNumber?: number,
+    sequenceNumber: number,
+    msgLength?: number,
+    segmentPayload?: ByteArray
+}
+
+export interface BtpPacket {
+    header: Header,
+    payload: BtpPacketPayload
+}
+
+export const enum HeaderBits {
+    HandshakeBit = 0b00000010,
+    ManagementMsg = 0b00000100,
+    AckMsg = 0b00010000,
+    EndSegment = 0b00100000,
+    BeginSegment = 0b10000000
+}
+
+export const enum Opcode {
+    HandshakeManagementOpcode = 0x6c,
+}
+
+const HANDSHAKE_HEADER = 0b01100101;
+
+var ver: Uint8Array = new Uint8Array();
+
+export class BtpCodec {
+
+    static decodeBtpHandshakeRequest(data: ByteArray): BtpHandshakeRequest {
+        const reader = new DataReader(data, Endian.Little);
+        return this.decodeRequestPayload(reader);
+    }
+
+    static decodeBtpPacket(data: ByteArray): BtpPacket {
+        const reader = new DataReader(data, Endian.Little);
+
+        return {
+            header: this.decodeHeader(reader),
+            payload: this.decodeBtpPacketPayload(reader)
+        };
+
+    }
+
+    static encodeBtpHandshakeResponse({ version, attMtu, windowSize }: BtpHandshakeResponse): ByteArray {
+
+        return ByteArray.concat(
+            new Uint8Array(HANDSHAKE_HEADER),
```

Comment:
Can we use a ByteArray class here too? I know it is an alias but so we do not mix stuff

Reply:
Updated